packages/haproxy/data_stream/log/elasticsearch/ingest_pipeline/default.yml

```
@@ -66,6 +66,7 @@ processors:
       patterns:
         - '%{WORD:http.request.method}%{SPACE}%{URIPATHPARAM:url.original}%{SPACE}HTTP/%{NUMBER:http.version}'
       ignore_missing: true
+      ignore_failure: true
```

Can we use on_failure: processor instead of ignore_failure!

```
      on_failure:
        - append:
           field: error.message
           value: 'Processor {{{_ingest.on_failure_processor_type}}} with tag fail-{{{_ingest.on_failure_processor_tag}}} in pipeline {{{_ingest.pipeline}}} failed with message: {{{_ingest.on_failure_message}}}'
```

---

Thanks @harnish-elastic, I had tried this, but the pipeline fails with the error message(which is similar to the already existing case), 
```
 "error": {
                "message": [
                    "Processor grok with tag fail- in pipeline default-1745303520704446129 failed with message: Provided Grok expressions do not match field value: [GET /public/index.php?s=/index/\\think\\app/testinvokefunction&function=call_user_func_array_test&vars[0]=md15&vars[1][]=Helloworld HTTP/1.1]"
                ]
            }
```
According to this [SDH comment](https://github.com/elastic/sdh-beats/issues/5509#issuecomment-2560974804), we want something which can handle the malformed URLs correctly without returning an error.

---

we can return the error if the format doesn't match as well continue the pipeline to perform other processors. So in this case if the malformed URL is not correct, we can throw error by `on_failure` processor as well we can keep the pipeline to perform the rest processors.

---

Also, if the other fields of the grok are not correct, on_failure should be there to return error for those values from the grok!